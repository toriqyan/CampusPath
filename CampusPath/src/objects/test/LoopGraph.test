# Tests that a graph can be created, four nodes and
# five edges that create a loop can be added, 
# and four nodes and five edges can be printed out.

# Create a graph
CreateGraph TheLordOfTheRing4

# Add four nodes
AddNode TheLordOfTheRing4 Frodo
AddNode TheLordOfTheRing4 Gandalf
AddNode TheLordOfTheRing4 Samwise
AddNode TheLordOfTheRing4 Arwin

# Add five edges
AddEdge TheLordOfTheRing4 Gandalf Frodo TheHobbit
AddEdge TheLordOfTheRing4 Frodo Samwise TheReturnOfTheKing
AddEdge TheLordOfTheRing4 Frodo Samwise TheTwoTowers
AddEdge TheLordOfTheRing4 Samwise Arwin TheFellowshipOfTheRing
AddEdge TheLordOfTheRing4 Arwin Gandalf TheFellowshipOfTheRing

# Print the nodes in the graph and the outgoing edges from Frodo, Gandalf, Arwin, Samwise
ListNodes TheLordOfTheRing4
ListChildren TheLordOfTheRing4 Frodo
ListChildren TheLordOfTheRing4 Gandalf
ListChildren TheLordOfTheRing4 Arwin
ListChildren TheLordOfTheRing4 Samwise