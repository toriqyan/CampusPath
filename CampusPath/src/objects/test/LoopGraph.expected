# Tests that a graph can be created, four nodes and
# five edges that create a loop can be added, 
# and four nodes and five edges can be printed out.

# Create a graph
created graph TheLordOfTheRing4

# Add four nodes
added node Frodo to TheLordOfTheRing4
added node Gandalf to TheLordOfTheRing4
added node Samwise to TheLordOfTheRing4
added node Arwin to TheLordOfTheRing4

# Add five edges
added edge TheHobbit from Gandalf to Frodo in TheLordOfTheRing4
added edge TheReturnOfTheKing from Frodo to Samwise in TheLordOfTheRing4
added edge TheTwoTowers from Frodo to Samwise in TheLordOfTheRing4
added edge TheFellowshipOfTheRing from Samwise to Arwin in TheLordOfTheRing4
added edge TheFellowshipOfTheRing from Arwin to Gandalf in TheLordOfTheRing4

# Print the nodes in the graph and the outgoing edges from Frodo, Gandalf, Arwin, Samwise
TheLordOfTheRing4 contains: Arwin Frodo Gandalf Samwise
the children of Frodo in TheLordOfTheRing4 are: Samwise(TheReturnOfTheKing) Samwise(TheTwoTowers)
the children of Gandalf in TheLordOfTheRing4 are: Frodo(TheHobbit)
the children of Arwin in TheLordOfTheRing4 are: Gandalf(TheFellowshipOfTheRing)
the children of Samwise in TheLordOfTheRing4 are: Arwin(TheFellowshipOfTheRing)